### PR TITLE
Add bridge extension validation

### LINKED ISSUE #101: Add `BridgeRejectObsoleteHeadersAndMessages` back
```rs
pub type SignedExtra = (
	CheckNonZeroSender<Runtime>,
	CheckSpecVersion<Runtime>,
	CheckTxVersion<Runtime>,
	CheckGenesis<Runtime>,
	CheckEra<Runtime>,
	CheckNonce<Runtime>,
	CheckWeight<Runtime>,
	ChargeTransactionPayment<Runtime>,
	BridgeRejectObsoleteHeadersAndMessages,  // this one
);
```

```rs
bridge_runtime_common::generate_bridge_reject_obsolete_headers_and_messages! {
	Call, AccountId,
	// Grandpa
	BridgePangoroGrandpa, BridgeRococoGrandpa, BridgeMoonbaseRelayGrandpa,
	// Messages
	BridgePangoroMessages, BridgePangolinParachainMessages, BridgePangolinParachainAlphaMessages,
	// Parachain
	BridgeRococoParachain, BridgeMoonbaseRelayParachain
}
```

### PR DESCRIPTION
**Pull request by boundless-forest**
Close #101

### CODE REVIEW

**Review by jiguantong — DISMISSED** [review]

File: runtime/crab/src/lib.rs [hunk 1]
```diff
@@ -144,14 +145,24 @@ where
...
 
-impl_self_contained_call!();
-
 /// The version information used to identify this runtime when compiled natively.
 #[cfg(feature = "std")]
 pub fn native_version() -> NativeVersion {
 	NativeVersion { runtime_version: VERSION, can_author_with: Default::default() }
 }
 
+impl_self_contained_call!();
```
Comment (AurevoirXavier) [inline runtime/crab/src/lib.rs:154]: BTW, I want to put it above the `sp_api::impl_runtime_apis! {`.

File: runtime/darwinia/src/lib.rs [hunk 2]
```diff
@@ -243,6 +231,18 @@ frame_benchmarking::define_benchmarks! {
 	[cumulus_pallet_xcmp_queue, XcmpQueue]
 }
 
+impl_self_contained_call!();
```
Comment (AurevoirXavier) [inline runtime/darwinia/src/lib.rs:234]: This

File: runtime/crab/src/lib.rs (hunk 1 above)
Comment (boundless-forest) [reply runtime/crab/src/lib.rs]: I made some format updates as you suggested. Please check it again.

File: runtime/crab/src/lib.rs [hunk 3]
```diff
@@ -243,6 +231,18 @@ frame_benchmarking::define_benchmarks! {
 	[cumulus_pallet_xcmp_queue, XcmpQueue]
 }
 
+impl_self_contained_call!();
```
Comment (AurevoirXavier) [inline runtime/crab/src/lib.rs:234]: This

**Review by AurevoirXavier — APPROVED** [review]

**Pull request merged**